litr/src/main/java/com/linkedin/android/litr/render/PassthroughSoftwareRenderer.java

```diff
@@ -49,33 +50,56 @@ public void renderFrame(@Nullable Frame frame, long presentationTimeNs) {
             Log.e(TAG, "Null or empty input frame provided");
             return;
         }
-        int tag = encoder.dequeueInputFrame(FRAME_WAIT_TIMEOUT);
-        if (tag >= 0) {
-            Frame outputFrame = encoder.getInputFrame(tag);
-            if (outputFrame == null) {
-                Log.e(TAG, "No input frame returned by an encoder, dropping a frame");
-                return;
-            }
 
-            ByteBuffer inputBuffer = frame.buffer.asReadOnlyBuffer();
-            inputBuffer.rewind();
-            outputFrame.buffer.put(inputBuffer);
+        ArrayList<Frame> outputFrames = new ArrayList<>();
+        ByteBuffer inputBuffer = null;
+        boolean isFirst = true;
+        boolean isContinue;
+        do {
+            isContinue = false;
+            int tag = encoder.dequeueInputFrame(FRAME_WAIT_TIMEOUT);
+            if (tag >= 0) {
+                Frame outputFrame = encoder.getInputFrame(tag);
+                if (outputFrame == null) {
+                    Log.e(TAG, "No input frame returned by an encoder, dropping a frame");
+                    return;
+                }
+
+                if (isFirst) {
+                    isFirst = false;
+                    inputBuffer = frame.buffer.asReadOnlyBuffer();
+                    inputBuffer.rewind();
+                }
+
+                int capacity = Math.min(outputFrame.buffer.remaining(), inputBuffer.remaining());
```

Comment:
Another way is to have an `if (outputFrame.buffer.remaining() < inputBuffer.remaining()` comparison here and do buffer copy (new logic) only when it is true. Otherwise, we can just copy directly, like we were doing before.